### ISSUE TITLE
Bug in NSGA-II in jMetal 5.8+: getting solutions from population during reproduction

### ISSUE THREAD

**Issue by GiovaniGuizzo** [jMetal/jMetal#402]
I was reading the code of NSGA-II and found out that, during reproduction, the solutions are taken sequentially from the population itself instead of the mating pool. See line 122:

https://github.com/jMetal/jMetal/blob/831d62d0bbf384e1770efc1bb6eef69ce0ce75b9/jmetal-algorithm/src/main/java/org/uma/jmetal/algorithm/multiobjective/nsgaii/NSGAII.java#L113-L125

I believe it is a bug, because the mating pool is the one containing the selected parents from the selection phase. Moreover, the indexes are taken from the mating pool, which is another indication that the reproduction is using the wrong list.

If it is indeed a bug (I might be wrong), the fix is straightforward: `matingPool.get(i+j)` instead of `population.get(i+j)` in line 122.

This code is the same for versions 5.7 and 5.9 too.

**Comment by ajnebro** [comment]
Thanks for reporting the bug  and the pull request.

**Comment by ajnebro** [comment]
In the NSGA-II version in the 6.0-SNAPSHOT release it doesn't appear (fortunately):

```java

  @Override
  public List<S> variate(List<S> population, List<S> matingPopulation) {
    int numberOfParents = crossover.getNumberOfRequiredParents();

    checkNumberOfParents(matingPopulation, numberOfParents);

    List<S> offspringPopulation = new ArrayList<>(offspringPopulationSize);
    for (int i = 0; i < matingPoolSize; i += numberOfParents) {
      List<S> parents = new ArrayList<>(numberOfParents);
      for (int j = 0; j < numberOfParents; j++) {
        parents.add(matingPopulation.get(i + j));
      }

      List<S> offspring = crossover.execute(parents);
```

**Comment by GiovaniGuizzo** [comment]
> In the NSGA-II version in the 6.0-SNAPSHOT release it doesn't appear (fortunately):

Yes, because this was the fix of my Pull Request #403. I don't know how to fix it in the jMetal 5.9 version, since it is just a tag.

**Comment by ajnebro** [comment]
Hmmm, I'm not sure ... let me see whether it is possible.

**Comment by GiovaniGuizzo** [comment]
I guess since the fix from my pull request was merged into the 5.10 release, this issue can be closed.
